### PR TITLE
[WIP] Fix linux build

### PR DESCRIPTION
**Pull request by IcanDivideBy0**
This is an atempt to fix the linux build.
Please note that I didn't wrote c in the last 10 years, so take this with care!
Also I have absolutely no idea what i'm doing with those CMake files...

With those fixes I managed to get the material editor running (but i'm havin a segfault when i'm trying to save).

Some minor changes are also necessary in `AltseedRHI` `libgd` and `zlib` submodules (mostly including cstring here and there).

The build did not terminate properly tho, I'm not sure what's the issue... (seems to miss msbuild but i don't even know what it is :/)

```
....

[ 98%] Built target Example_OpenAL
[100%] Built target Example_CustomLoader

mono ./build/nuget.exe restore Dev/Editor/Effekseer.sln
Could not load file or assembly 'System.ComponentModel.Composition, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b77a5c561934e089' or one of its dependencies.
System.IO.FileNotFoundException: Could not load file or assembly 'System.ComponentModel.Composition, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b77a5c561934e089' or one of its dependencies.
File name: 'System.ComponentModel.Composition, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b77a5c561934e089'
  at NuGet.CommandLine.Program.MainCore (System.String workingDirectory, System.String[] args) [0x000d8] in <f495a477196947558735199950d701a7>:0 

"msbuild" Dev/Editor/EffekseerCore/EffekseerCore.csproj /t:build /p:Configuration=Release /p:Platform=x64
/bin/sh: 1: msbuild: not found
Failed "msbuild" Dev/Editor/EffekseerCore/EffekseerCore.csproj /t:build /p:Configuration=Release /p:Platform=x64
Traceback (most recent call last):
  File "build.py", line 82, in <module>
    call('"' + msbuild_path + '"' + ' Dev/Editor/EffekseerCore/EffekseerCore.csproj /t:build /p:Configuration=Release /p:Platform=x64')
  File "build.py", line 15, in call
    raise Exception
Exception

```

more infos:
```
$ uname -a
Linux debian 4.19.0-8-amd64 #1 SMP Debian 4.19.98-1 (2020-01-26) x86_64 GNU/Linux
$ mono --version
Mono JIT compiler version 5.18.0.240 (Debian 5.18.0.240+dfsg-3 Wed Apr 17 16:37:36 UTC 2019)
$ g++ --version
g++ (Debian 8.3.0-6) 8.3.0
$ cmake --version
cmake version 3.13.4
```

Also, as a footnote, let me know if you have any Rust integration initiative, I'll probaly work on this if I can get the editor runningon linux.

### CODE REVIEW

File: Dev/Cpp/Viewer/GUI/efk.GUIManager.cpp [hunk 1]
```diff
@@ -1109,7 +1109,7 @@ namespace efk
 		return Vec2(v.x, v.y);
 	}
 
-	void GUIManager::SetNextWindowPos(const Vec2& pos, Cond cond, const const Vec2& pivot)
```
Comment (IcanDivideBy0) [inline Dev/Cpp/Viewer/GUI/efk.GUIManager.cpp:1112]: I think this particular change is not correct, tell me how I can escape the `duplicate ‘const’` error here ?

Comment (durswd) [reply Dev/Cpp/Viewer/GUI/efk.GUIManager.cpp:1112]: It is a my mistake. Please remove first const.

**Comment by durswd** [comment]
Please change if(UNIX) into if(NOT APPLE AND UNIX)
x11 is not found on Mac, so CI fails

**Comment by durswd** [comment]
@IcanDivideBy0 
mono error -> I will check it. It seem that dll is not found

**Comment by durswd** [comment]
AltseedRHI -> I can merge a PR
But I cannot merge a PR on zlib and libgd.
Please send a PR into each repositories.

**Comment by durswd** [comment]
I'm trying to compile C# on linux... It seem that it is better to change .net version. So please wait a few days

**Comment by IcanDivideBy0** [comment]
Ok! Didn't re-opened the project since PR. Feel free to go ahead!

**Comment by durswd** [comment]
I rewrite C# project and I will merge this PR

**Comment by durswd** [comment]
I fix errors

**Comment by durswd** [comment]
At least, Effekseer now can be built on Docker

https://github.com/effekseer/Effekseer/blob/master/Docker/Ubuntu/Dockerfile

**Pull request merged**